### PR TITLE
Fix staking.info

### LINKED ISSUE #1060: props for derive.staking.info not updated
An issue appeared in `apps` and is related to `derive.staking.info`. Updates are not pushed through the props, so that the info don't update unless the page is reloaded.

I've tracked down the issue and it happens since v0.82 https://github.com/polkadot-js/api/pull/987

How to reproduce:
- navigate to staking>actions
- change the reward destination
- once the tx succeeds, the reward destination isn't updated
- refresh the page or change tab and get back -> the destination is updated.

### LINKED ISSUE #1092: api.queryMulti should cache values
```javascript
api.queryMulti([
  [api.query.session.nextKeyFor, controllerId],
  [api.query.staking.ledger, controllerId],
], callback)
```

First result: [something1, something2],

**Expected:** 

second result: [something1, something3]

**Actual:**

second result: [null, something3]

### PR DESCRIPTION
**Pull request by amaurymartiny**
fixes #1060 
fixes #1092

### CODE REVIEW

File: packages/api-derive/src/staking/info.ts [hunk 1]
```diff
@@ -83,21 +83,23 @@ function withStashController (api: ApiInterface$Rx, accountId: AccountId, contro
...
+        [api.query.staking.validators, stashId]
+      ])
+    ]) as any as Observable<[BN, BlockNumber, [Option<Keys | SessionKey>, Option<StakingLedger>, [Array<AccountId>], RewardDestination, Exposure, [ValidatorPrefs]]]>
   ).pipe(
-    map(([eraLength, bestNumber, [nextKeyFor, _stakingLedger, [nominators], rewardDestination, stakers, [validatorPrefs]]]) => {
-      const stakingLedger = _stakingLedger.unwrapOr(null) || undefined;
+    map(([eraLength, bestNumber, [nextKeyFor, _stakingLedger, _nominators, rewardDestination, stakers, _validatorPrefs]]) => {
+      const nominators = _nominators ? _nominators[0] : undefined;
+      const validatorPrefs = _validatorPrefs ? _validatorPrefs[0] : undefined;
+      const stakingLedger = _stakingLedger ? _stakingLedger.unwrapOr(undefined) : undefined;
```
Comment (amaurymartiny) [inline packages/api-derive/src/staking/info.ts]: So the errors came from here, undefined & null...

File: packages/api-derive/src/util/drr.ts [hunk 2]
```diff
@@ -12,6 +15,11 @@ import { Observable } from 'rxjs';
  */
 export const drr = () => <T>(source$: Observable<T>): Observable<T> =>
   source$.pipe(
+    catchError(err => {
+      l.error(err);
```
Comment (amaurymartiny) [inline packages/api-derive/src/util/drr.ts]: ... and were never logged inside the Observable loops. I added an explicit log here

File: packages/api-derive/test/e2e/promise.spec.ts [hunk 3]
```diff
@@ -171,5 +167,45 @@ describe.skip('derive e2e', () => {
         done();
       });
     });
+
+    // FIXME This test doesn't pass
+    // https://github.com/polkadot-js/api/issues/1060
```
Comment (Stefie) [inline packages/api-derive/test/e2e/promise.spec.ts]: Can we please not add it, then? Or just put it in an PR for fixing the issue mentioned in the comment?

I'm just working on making all e2e tests pass again at least in 1.0 and master.
I fixed all the derive ones in https://github.com/polkadot-js/api/pull/1084 and now I'm on the contract tests, the child storage tests and some other ones that were failing.

If we add tests, they should pass or the methods should be fixed if they're broken.

Comment (jacogr) [reply packages/api-derive/test/e2e/promise.spec.ts]: Making that test pass is the whole point of this PR - so the comment will be removed as soon as it does.

Comment (Stefie) [reply packages/api-derive/test/e2e/promise.spec.ts]: Ah, that wasn’t clear from the title, I thought it was primarily about logging/ error handling. Looking forward to it, then!

Comment (jacogr) [reply packages/api-derive/test/e2e/promise.spec.ts]: The initial fix part was reverted when we realised we need more work and it is not quite the correct solution. (The test actually did pass on it, however it had some effects on API users which we are not keen on)

**Comment by amaurymartiny** [comment]
OK frankenstein now passes

**Comment by Stefie** [comment]
One of the tests actually fails on 1.0:
Screenshot 2019-07-10 at 18 45 27

**Comment by jacogr** [comment]
@Stefie Ok, it really has been a long day for me, but I cannot quite seem how the hell that happens - so the failing test is Frankenstein line 171 onwards, but is shows a failure on the previous test line. Something is very weird.

**Pull request merged**